src/webview/common/sanitize.ts

```diff
@@ -0,0 +1,25 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+/**
+ * Builds a sanitized compponent name from a component folder full path or
+ * a Git-repo URL
+ *
+ * @param pathOrURL an absolute path ot a Git-repo URL
+ * @returns sanitized component name
+ */
+export function buildSanitizedComponentName(pathOrURL?: string): string {
```

Comment:
I realize this mess of regex is something that I wrote and should have tested more thoroughly. Here is a case that isn't sanitized properly: https://github.com/datho7561/12345--painful--to---manage....git...repo...name

Reply:
It becomes `-painful-to-manage-git-repo-name`, which isn't valid.

Minimal reproducer is `1234-a`:
- `1234` gets reduced to `` in the last step, leaving `-a`. The leading `-` is invalid

Reply:
Yeah, Hope it's fixed now, at least for your examples now it returns  `a` and `painful-to-manage-git-repo-name`